packages/interactivity/src/utils.js

```diff
@@ -53,6 +58,51 @@ export function useSignalEffect( callback ) {
 	}, [] );
 }
 
+/**
+ * Returns the passed function wrapped with the current scope so it is
+ * accessible whenever the function runs. This is primarily to make the scope
+ * available inside hook callbacks.
+ *
+ * @param {Function} func The passed function.
+ * @return {Function} The wrapped function.
+ */
+const withScope = ( func ) => {
+	const scope = getScope();
+	return ( ...args ) => {
+		setScope( scope );
+		const output = func( ...args );
+		resetScope();
+		return output;
```

Comment:
Should we try/catch here? Something like:

```suggestion
		setScope( scope );
    try {
		  const output = func( ...args );
		  return output;
    } catch (e) {
      throw e;
    } finally {
		  resetScope();
    }
```

Reply:
Oh, right! Absolutely. 😄

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/57805/commits/966f02d65a9ba7d869097ab39683566c117f81a9.